File: src/main/java/com/ggardet/doctor/spi/exception/SpecialtyNotFoundException.java
package com.ggardet.doctor.spi.exception;

import com.ggardet.core.exception.InternalNotFoundException;

import java.util.UUID;

public class SpecialtyNotFoundException extends InternalNotFoundException {
    public SpecialtyNotFoundException(final UUID specialtyId) {
        super(String.format("Specialty not found: %s", specialtyId));
    }
    public SpecialtyNotFoundException(final String specialtyName) {
        super(String.format("Specialty not found: %s", specialtyName));
    }
}

File: src/main/kotlin/com/ggardet/doctor/spi/exception/SpecialtyNotFoundException.kt
package com.ggardet.doctor.spi.exception

import com.ggardet.core.spi.exception.InternalNotFoundException
import java.util.*

class SpecialtyNotFoundException : InternalNotFoundException {
    constructor(specialtyId: UUID) : super("Specialty not found: $specialtyId")
    constructor(specialtyName: String) : super("Specialty not found: $specialtyName")
}